kmath-trajectory/src/commonMain/kotlin/space/kscience/kmath/trajectory/dubins/DubinsPathFactory.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2018-2021 KMath contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package space.kscience.kmath.trajectory.dubins
+
+import space.kscience.kmath.geometry.Euclidean2DSpace.distanceTo
+import space.kscience.kmath.geometry.Line2D
+import space.kscience.kmath.geometry.Vector2D
+import space.kscience.kmath.trajectory.segments.Arc
+import space.kscience.kmath.trajectory.segments.LineSegment
+import space.kscience.kmath.trajectory.segments.components.Circle
+import space.kscience.kmath.trajectory.segments.components.Pose2D
+import space.kscience.kmath.trajectory.segments.length
+import space.kscience.kmath.trajectory.segments.theta
+import kotlin.math.acos
+import kotlin.math.cos
+import kotlin.math.sin
+
+public class DubinsPathFactory(
+    private val base: Pose2D,
+    private val direction: Pose2D,
+    private val turningRadius: Double
+) {
+
+    public val all: List<DubinsPath> get() = listOfNotNull(rlr, lrl, rsr, lsl, rsl, lsr)
+    public val shortest: DubinsPath get() = all.minByOrNull { it.length }!!
+    public operator fun get(type: DubinsPath.TYPE): DubinsPath? = all.find { it.type == type }
+
+    public val rlr: DubinsPath?
```

Comment:
does it make sense to add lazy delegate to allow caching?

Reply:
Reworked, DubinsPath contains factory functions now